### ISSUE TITLE
如何加载训练好的权重做dev和test？

### ISSUE THREAD

**Issue by diaoyudiaochan** [liuxiyang641/HVFormer#1]
up主您好，请问如何修改run_mre.sh文件使其可以进行dev和test？能否在方便的时候给个样例呢，谢谢up主！

**Comment by liuxiyang641** [comment]
你好，可以尝试加入命令`--do_test`和`-load_path="your checkpoint file path"`来在test set上进行测试。单独进行dev数据集上的测试可能需要修改代码。

**Comment by diaoyudiaochan** [comment]
谢谢作者及时的回复，不好意思我的问题还是没有解决。我将run_mre.sh的内容修改如下所示：“
#!/usr/bin/env bash
source ~/.bashrc

DATASET_NAME="mnre"
BERT_NAME="bert-base-uncased"
VIT_NAME="openai/clip-vit-base-patch32"
EXP_NAME="test"

CUDA_VISIBLE_DEVICES=0 python -u run.py \
        --model_name="bert-vit-inter-re" \
        --experiment_name=${EXP_NAME} \
        --vit_name=$VIT_NAME \
        --dataset_name=${DATASET_NAME} \
        --bert_name=${BERT_NAME} \
        --num_epochs=1 \
        --batch_size=32 \
        --lr=1e-5 \
        --warmup_ratio=0.06 \
        --eval_begin_epoch=1 \
        --max_seq=80 \
        --prompt_len=4 \
        --aux_size=128 \
        --rcnn_size=64 \
        --do_test \
        --save_path="ckpt" \
        --write_path="logs" \
        --device='cuda' \
        --num_workers=10 \
        --load_path="\train_mnre_2024_07_30_11_20_00" \
”，ckpt文件夹下生成的文件是没有后缀名的，如下所示：“
train_mnre_2024_07_30_11_20_00
”，这样配置之后用bash run_mre.sh,输出如下所示：“
Namespace(aux_size=128, batch_size=32, bert_name='bert-base-uncased', crf_lr=0.05, dataset_name='mnre', device='cuda', do_predict=False, do_test=True, do_train=False, eval_begin_epoch=1, experiment_name='\\train_mnre_2024_07_30_11_20_00', ignore_idx=0, load_path='ckpt/\\train_mnre_2024_07_30_11_20_00', log_mode='logger', lr=1e-05, max_seq=80, model_name='bert-vit-inter-re', notes='', num_epochs=1, num_workers=10, prompt_len=4, prompt_lr=0.0003, rcnn_size=64, save_path='ckpt', seed=1, vit_name='openai/clip-vit-base-patch32', warmup_ratio=0.06, write_path='logs/\\train_mnre_2024_07_30_11_20_00')
”。没有任何输出，请问这个问题该怎么解决呢。谢谢作者！！！

**Comment by liuxiyang641** [comment]
试试`--load_path="train_mnre_2024_07_30_11_20_00"`，不需要加`\`

**Comment by diaoyudiaochan** [comment]
试过了，换成#!/usr/bin/env bash
source ~/.bashrc

DATASET_NAME="mnre"
BERT_NAME="bert-base-uncased"
VIT_NAME="openai/clip-vit-base-patch32"
EXP_NAME="test"

CUDA_VISIBLE_DEVICES=0 python -u run.py \
        --model_name="bert-vit-inter-re" \
        --experiment_name=${EXP_NAME} \
        --vit_name=$VIT_NAME \
        --dataset_name=${DATASET_NAME} \
        --bert_name=${BERT_NAME} \
        --num_epochs=1 \
        --batch_size=32 \
        --lr=1e-5 \
        --warmup_ratio=0.06 \
        --eval_begin_epoch=1 \
        --max_seq=80 \
        --prompt_len=4 \
        --aux_size=128 \
        --rcnn_size=64 \
        --do_test \
        --save_path="ckpt" \
        --write_path="logs" \
        --device='cuda' \
        --num_workers=10 \
        --load_path="train_mnre_2024_07_31_16_45_33" ，输出也是只有“Namespace(aux_size=128, batch_size=32, bert_name='bert-base-uncased', crf_lr=0.05, dataset_name='mnre', device='cuda', do_predict=False, do_test=True, do_train=False, eval_begin_epoch=1, experiment_name='train_mnre_2024_07_31_16_45_33', ignore_idx=0, load_path='ckpt/train_mnre_2024_07_31_16_45_33', log_mode='logger', lr=1e-05, max_seq=80, model_name='bert-vit-inter-re', notes='', num_epochs=1, num_workers=10, prompt_len=4, prompt_lr=0.0003, rcnn_size=64, save_path='ckpt', seed=1, vit_name='openai/clip-vit-base-patch32', warmup_ratio=0.06, write_path='logs/train_mnre_2024_07_31_16_45_33')”。不知道为什么没有运行测试

**Comment by liuxiyang641** [comment]
抱歉，漏了一个命令，加入`--do_train`，它和`--do_test`是一起的，如果没有`--do_train`也不会进行后续的测试。参考`bert_vit_re_train.py`文件里的`train(self)`函数中的代码：
```python
if self.args.do_test:
    self.logger.info("***** Start testing without training *****")
    self.test(0)
    return
```

**Comment by diaoyudiaochan** [comment]
谢谢作者的回复！！！加了--do_train之后在log文件夹生成了一个同名的csv文件，同时原log文件中记录了测试的结果。这下应该是份完整的测试流程了吧？再次感谢作者的快速回复，祝作者paper多多～

**Comment by liuxiyang641** [comment]
祝好
